File: leetcode/1.go
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	var x int
	var y int

	for i := 0; i < len(nums); i++ {

		cv := nums[i] // current value

		if v, ok := m[target-cv]; ok {
			x = v
			y = i

			// only one solution
			break
		}
		m[cv] = i // store index
	}

	return []int{x, y}
}

File: leetcode/python/1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        # only one solution
        for idx, v in enumerate(nums):
            if v not in d:
                d[target - v] = idx
            else:
                return [d[v], idx]
